pkg/controller/account/account_controller.go

```diff
@@ -158,6 +158,18 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 		return reconcile.Result{}, err
 	}
 
+	// Add finalizer to non-sts account cr
+	if !currentAcctInstance.Spec.ManualSTSMode {
+		if !controllerutils.Contains(currentAcctInstance.GetFinalizers(), awsv1alpha1.AccountFinalizer) {
```

Comment:
could this check actually happen in the addFinalizer function itself? So we could just call it here, and it would ensure the finalizer is only added if not there already?